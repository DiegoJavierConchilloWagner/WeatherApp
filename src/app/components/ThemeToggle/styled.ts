import styled from 'styled-components';

export const StyledThemeToggle = styled.div`
	position: absolute;
	right: 13%;
	top: 5%;

	@media (max-width: 768px) {
		right: 40%;
		top: 0.7rem;
	}
`;
