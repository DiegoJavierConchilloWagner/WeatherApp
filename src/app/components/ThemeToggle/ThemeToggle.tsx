/* eslint-disable jsx-a11y/tabindex-no-positive */
import { useState } from 'react';

import { useDispatch, useSelector } from 'react-redux';
import Switch from 'react-switch';

import { light, dark } from '@styled/themes';
import { ThemeModes } from '@common/index';
import { setThemeMode } from '@redux/actions';
import { RootState } from '@redux/store/store';
import { WeatherIcon } from '../WeatherIcon/WeatherIcon';

import { StyledThemeToggle } from './styled';

export const ThemeToggle = () => {
	const dispatch = useDispatch();
	const [checked, setChecked] = useState<boolean>(true);

	const { themeMode } = useSelector((state: RootState) => state.uiInterface);

	const handleCheck = () => {
		setChecked(!checked);
		checked === true ? dispatch(setThemeMode(ThemeModes.LIGHT)) : dispatch(setThemeMode(ThemeModes.DARK));
	};

	return (
		<StyledThemeToggle>
			<Switch
				tabIndex={6}
				onChange={handleCheck}
				checked={checked}
				height={40}
				width={80}
				borderRadius={45}
				offColor={themeMode === 'LIGHT' ? light.colors.switch.secondary : dark.colors.switch.secondary}
				onColor={themeMode === 'LIGHT' ? light.colors.switch.secondary : dark.colors.switch.secondary}
				offHandleColor={themeMode === 'LIGHT' ? light.colors.switch.primary : dark.colors.switch.primary}
				onHandleColor={themeMode === 'LIGHT' ? light.colors.switch.primary : dark.colors.switch.primary}
				uncheckedIcon={
					<div
						style={{
							display: 'flex',
							justifyContent: 'center',
							alignItems: 'center',
							textAlign: 'center',
							height: '100%',
							fontSize: 25,
							color: 'white',
							paddingRight: 2,
							fontFamily: 'Poppins'
						}}>
						<WeatherIcon id={800} icon='01d' size={30} isLoading={false} />
					</div>
				}
				checkedIcon={
					<div
						style={{
							display: 'flex',
							justifyContent: 'center',
							alignItems: 'center',
							height: '100%',
							fontSize: 25,
							color: 'white',
							paddingRight: 2,
							fontFamily: 'Poppins'
						}}>
						<WeatherIcon id={800} icon='01n' size={30} isLoading={false} />
					</div>
				}
			/>
		</StyledThemeToggle>
	);
};
