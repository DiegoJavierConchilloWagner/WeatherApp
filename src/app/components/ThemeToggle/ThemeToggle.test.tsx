import renderWithThemeAndProvider from '@common/utils/renderWhitThemeAndProvider';
import { ThemeToggle } from './ThemeToggle';

describe('Example ThemeToggle', () => {
	it('should render correcty', () => {
		const wrapper = renderWithThemeAndProvider(<ThemeToggle />);

		expect(wrapper).toMatchSnapshot();
	});
});
