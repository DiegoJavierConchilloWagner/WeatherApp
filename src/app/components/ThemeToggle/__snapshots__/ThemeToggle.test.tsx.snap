// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example ThemeToggle should render correcty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: absolute;
  right: 13%;
  top: 5%;
}

@media (max-width:768px) {
  .c0 {
    right: 40%;
    top: 0.7rem;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 20px; transition: opacity 0.25s; user-select: none;"
        >
          <div
            class="react-switch-bg"
            style="height: 40px; width: 80px; margin: 0px; position: relative; background: rgb(13, 15, 64); border-radius: 45px; cursor: pointer; transition: background 0.25s;"
          >
            <div
              style="height: 40px; width: 42px; position: relative; opacity: 1; pointer-events: none; transition: opacity 0.25s;"
            >
              <div
                style="display: flex; justify-content: center; align-items: center; height: 100%; font-size: 25px; color: white; padding-right: 2px; font-family: Poppins;"
              >
                <img
                  alt="weather"
                  height="30"
                  src="SvgrURL"
                  width="30"
                />
              </div>
            </div>
            <div
              style="height: 40px; width: 42px; position: absolute; opacity: 0; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
            >
              <div
                style="display: flex; justify-content: center; align-items: center; text-align: center; height: 100%; font-size: 25px; color: white; padding-right: 2px; font-family: Poppins;"
              >
                <img
                  alt="weather"
                  height="30"
                  src="SvgrURL"
                  width="30"
                />
              </div>
            </div>
          </div>
          <div
            class="react-switch-handle"
            style="height: 38px; width: 38px; background: rgb(131, 194, 218); display: inline-block; cursor: pointer; border-radius: 44px; position: absolute; transform: translateX(41px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
          />
          <input
            aria-checked="true"
            checked=""
            role="switch"
            style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
            tabindex="6"
            type="checkbox"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: absolute;
  right: 13%;
  top: 5%;
}

@media (max-width:768px) {
  .c0 {
    right: 40%;
    top: 0.7rem;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 20px; transition: opacity 0.25s; user-select: none;"
      >
        <div
          class="react-switch-bg"
          style="height: 40px; width: 80px; margin: 0px; position: relative; background: rgb(13, 15, 64); border-radius: 45px; cursor: pointer; transition: background 0.25s;"
        >
          <div
            style="height: 40px; width: 42px; position: relative; opacity: 1; pointer-events: none; transition: opacity 0.25s;"
          >
            <div
              style="display: flex; justify-content: center; align-items: center; height: 100%; font-size: 25px; color: white; padding-right: 2px; font-family: Poppins;"
            >
              <img
                alt="weather"
                height="30"
                src="SvgrURL"
                width="30"
              />
            </div>
          </div>
          <div
            style="height: 40px; width: 42px; position: absolute; opacity: 0; right: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
          >
            <div
              style="display: flex; justify-content: center; align-items: center; text-align: center; height: 100%; font-size: 25px; color: white; padding-right: 2px; font-family: Poppins;"
            >
              <img
                alt="weather"
                height="30"
                src="SvgrURL"
                width="30"
              />
            </div>
          </div>
        </div>
        <div
          class="react-switch-handle"
          style="height: 38px; width: 38px; background: rgb(131, 194, 218); display: inline-block; cursor: pointer; border-radius: 44px; position: absolute; transform: translateX(41px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
        />
        <input
          aria-checked="true"
          checked=""
          role="switch"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
          tabindex="6"
          type="checkbox"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
